agenta-cli/agenta/sdk/middleware/auth.py

```diff
@@ -0,0 +1,111 @@
+from typing import Callable, Optional
+from uuid import UUID
+from json import dumps
+from traceback import format_exc
+
+import httpx
+from starlette.middleware.base import BaseHTTPMiddleware
+from fastapi import FastAPI, Request, HTTPException
+
+from agenta.sdk.utils.logging import log
+from agenta.sdk.middleware.cache import TTLLRUCache
+
+
+class Deny(HTTPException):
+    def __init__(self) -> None:
+        super().__init__(status_code=401, detail="Unauthorized")
+
+
+cache = TTLLRUCache(capacity=512, ttl=15 * 60)  # 15 minutes
```

Comment:
let's have the cache value as env var so we can easily adjust it

Reply:
Fixed.